playbook/app/pb_kits/playbook/pb_icon/_icon.tsx

```diff
@@ -40,63 +42,119 @@ type IconProps = {
   spin?: boolean,
 } & GlobalProps
 
+const iconSizeMap = {
+  base: 16,
+  xs: .75,
+  sm: .875,
+  lg: 1.25
+}
+
+const getSvgDimensions = (aspectRatio = '1:1', size: string): {width: number, height: number} | null => {
+  if (!size) return null
+  const aspect = aspectRatio.split(':')
+
+  const scale = size.toLowerCase().endsWith('x') ?
+    parseInt(size.replace(/\x/i, '')) :
+    iconSizeMap[size as IconSizeNames]
+
+  return {
+    width: (iconSizeMap.base * scale) * parseInt(aspect[0]),
+    height: (iconSizeMap.base * scale) * parseInt(aspect[1])
+  }
+}
+
 const flipMap = {
-  horizontal: 'fa-flip-horizontal',
-  vertical: 'fa-flip-vertical',
-  both: 'fa-flip-horizontal fa-flip-vertical',
-  none: ""
+  fa: {
+    horizontal: 'fa-flip-horizontal',
+    vertical: 'fa-flip-vertical',
+    both: 'fa-flip-horizontal fa-flip-vertical',
+    none: ''
+  },
+  svg: {
+    horizontal: 'flip_horizontal',
+    vertical: 'flip_vertical',
+    both: 'flip_horizontal flip_vertical',
+    none: ''
+  }
+}
+const pulseMap = {
+  fa: 'fa-pulse',
+  svg: 'pulse'
+}
+const spinMap = {
+  fa: 'fa-spin',
+  svg: 'spin'
+}
+const rotateMap = {
+  fa: {
+    90: 'fa-rotate-90',
+    180: 'fa-rotate-180',
+    270: 'fa-rotate-270'
+  },
+  svg: {
+    90: 'rotate_90',
+    180: 'rotate_180',
+    270: 'rotate_270'
+  }
```

Comment:
Why are we adding so many things that are already working in font awesome. The idea is to USE the FA stuff as is and extend our new icons to render as if they are coming from an FA library (Minus the FA Dash Initial Class)

Reply:
This has been all removed, using the FA styles works